package/psyneulinkviewer/conda.py

```diff
@@ -0,0 +1,78 @@
+import platform
+import os
+import sys
+import subprocess
+import logging
+import configuration
+import wget
+import platform
+from setuptools import setup, find_packages
+from setuptools.command.install import install
+from packaging.version import Version
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+def activate_env():
+    logging.info("Creating conda ")
+    subprocess.run(configuration.create_env, shell=True)
+
+    logging.info("Activating conda ")
+    subprocess.run(configuration.activate_env, shell=True)
+
+def install_conda():
+    if os.name == 'posix':
+        bash_file = wget.download(configuration.linux_conda_bash, out="psyneulinkviewer")
+    elif platform.system() == 'Darwin':
+        bash_file = wget.download(configuration.mac_conda_bashf)
+
+    logging.info("Installing conda %s", bash_file)
+    logging.info(bash_file)
+    subprocess.run("chmod +x " + bash_file, shell=True)
+    subprocess.run(bash_file + " -b -u -p ~/miniconda3", shell=True)
+
+    activate_env()
+    logging.info("Clean up ")
+    subprocess.run("rm -rf " + bash_file, shell=True)
+
+
+def check_conda_installation():
+    try:
+        conda_version = subprocess.run(
```

Comment:
@jrmartin this will not work if conda is not installed in the user machine.
If an exception of type FileNotFoundError is detected, then in the exception we need to handle this part of the installation.

Reply:
```
(base) ddelpiano@usop:~$ python
Python 3.12.2 | packaged by Anaconda, Inc. | (main, Feb 27 2024, 17:35:02) [GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import subprocess
>>> subprocess.run(['dario', '--version'])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ddelpiano/miniconda3/lib/python3.12/subprocess.py", line 548, in run
    with Popen(*popenargs, **kwargs) as process:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ddelpiano/miniconda3/lib/python3.12/subprocess.py", line 1026, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/home/ddelpiano/miniconda3/lib/python3.12/subprocess.py", line 1953, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'dario'
```

Reply:
dario is just an example of a binary that does not exist in my machine, if the user that is installtin this never installed conda in its machine this part will fail.

Reply:
@ddelpiano yeah, good catch, fixed and tested to avoid this situation.